pkg/collection/entry/entry.go

```diff
@@ -50,11 +51,17 @@ func (e *Entry) MarshalBinary() ([]byte, error) {
 }
 
 // UnmarshalBinary implements encoding.BinaryUnmarshaler
-func (e *Entry) UnmarshalBinary(b []byte) error {
-	if len(b) != serializedDataSize {
+func (e *Entry) UnmarshalBinary(b []byte, toDecrypt bool) error {
```

Comment:
`UnmarshalBinary implements encoding.BinaryUnmarshaler`

not if u introduce an argument.
i think it is ok to pass the `toDecrypt` flag within the entry  (as an unserialised or even serialised field).

Reply:
Removed the flag so that it remains like the old code.